File: go/shared/input.go
package shared

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func LoadInput() []string {
	file, err := os.Open("./input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	results := []string{}

	for scanner.Scan() {
		results = append(results, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	return results
}

func PrintRed(msg string) {
	fmt.Println("\x1b[37;41;1m", msg, "\x1b[0m")
}

func PrintGreen(msg string) {
	fmt.Println("\x1b[32;1m", msg, "\x1b[0m")
}

func PrintYellow(msg string) {
	fmt.Println("\x1b[33;1m", msg, "\x1b[0m")
}

File: ruby/shared/input.rb

def read_input()
  File.readlines('./input.txt')
end

def print_red(msg)
  puts red(msg)
end

def print_green(msg)
  puts green(msg)
end

def print_yellow(msg)
  puts yellow(msg)
end

def red(msg)
  "\x1b[37;41;1m#{msg}\x1b[0m"
end

def green(msg)
  "\x1b[32;1m#{msg}\x1b[0m"
end

def yellow(msg)
  "\x1b[33;1m#{msg}\x1b[0m"
end

def test()
  throw "read input failured" unless read_input().size == 5

  print_red("red");
  print_green("green");
  print_yellow("yellow");
end